### PR TITLE
[expat] Update library to 2.2.9

### PR DESCRIPTION
**Pull request by ehsan-mohammadi**
Update `expat` library to 2.2.9 version.

### CODE REVIEW

**Comment by ehsan-mohammadi** [comment]
It gives error on some windows triplets:

- ### `x86-windows`:
> REGRESSION: io2d:x86-windows
 REGRESSION: apr-util:x86-windows
 REGRESSION: gdal:x86-windows
 REGRESSION: readosm:x86-windows

- ### `x64-windows`:
> REGRESSION: io2d:x64-windows
 REGRESSION: apr-util:x64-windows
 REGRESSION: gdal:x64-windows
 REGRESSION: readosm:x64-windows
 REGRESSION: gtk:x64-windows

- ### `x64-windows-static`:
> REGRESSION: readosm:x64-windows-static
 REGRESSION: apr-util:x64-windows-static
 REGRESSION: gdal:x64-windows-static

- ### `arm64-windows`:
> REGRESSION: readosm:arm64-windows

**Comment by LilyWangL** [comment]
Pinging @ehsan-mohammadi for response. Is work still being done for this PR?

File: ports/expat/portfile.cmake [hunk 1]
```diff
@@ -1,13 +1,13 @@
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
-message(FATAL_ERROR "${PORT} does not currently support UWP")
+    message(FATAL_ERROR "${PORT} does not currently support UWP")
 endif()
 
 include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/expat/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/expat/portfile.cmake [hunk 2]
```diff
@@ -1,13 +1,13 @@
 if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
-message(FATAL_ERROR "${PORT} does not currently support UWP")
+    message(FATAL_ERROR "${PORT} does not currently support UWP")
```
Comment (LilyWangL) [inline ports/expat/portfile.cmake]: Please use `vcpkg_fail_port_install(ON_TARGET "UWP")` to description this port doesn't support UWP. And please modify _VCPKG_PATH/scripts/ci.baseline.txt_ to account for the UWP failures.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/expat/portfile.cmake (hunk 2 above)
Comment (ehsan-mohammadi) [reply ports/expat/portfile.cmake]: @LilyWangL Thanks for helping me! I resolved these conversations.

**Comment by LilyWangL** [comment]
@ehsan-mohammadi  Currently, this PR fails on Windows due to:
```
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
EXPAT_LIB
    linked by target "io2d_cairo_win32" in directory C:/vsts/_work/4/s/buildtrees/io2d/src/3b5b5fc338-1775a03870/P0267_RefImpl/P0267_RefImpl/cairo/win32
```
Ports such as io2d and gdal are dependent on expat. Due to this update, these build failed. Could you continue to resolve this PR?

**Comment by LilyWangL** [comment]
Pinging @ehsan-mohammadi for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**